Add clarification docs for differences with other tools (#530)
kubernetes-sigs/kwok: site/content/en/_index.md

- `kwok` is the cornerstone of this project, responsible for simulating the lifecycle of fake nodes, pods, and other Kubernetes API resources.
- `kwokctl` is a CLI tool designed to streamline the creation and management of clusters, with nodes simulated by `kwok`.

### What's the difference with `kubemark`

> [kubemark] is a kubelet that does not actually run a container, that behaviors are exactly the same as kubelet,
> so simulating a large number of nodes and pods requires a lot of memory.

Reviewer: ```suggestion
[kubemark] is a kubelet that does not actually run a container. Its behavior is exactly the same as kubelet,
which means that simulating a large number of nodes and pods requires a lot of memory.
```

> `kwok` just simulates the behavior of the node, so it can simulate a large number of nodes and pods with very little memory.

Reviewer: ```suggestion
`kwok`, however, simply simulates the behavior of the node. As a result, it can simulate a large number of nodes and pods using very little memory.
```

### What's the difference with `kind`

> [kind] is run Kubernetes in Docker that is a real cluster.

Reviewer: ```suggestion
[kind] runs Kubernetes in Docker, creating a real cluster.
```

> `kwokctl` is can be used as an alternative to [kind] in some scenarios where you don’t need to actually run the Pod.

Reviewer: ```suggestion
`kwokctl` can be used as an alternative to [kind] in some scenarios where you don’t need to actually run the Pod.
```
